pylint/lint/report_functions.py

```diff
@@ -16,33 +22,45 @@ def report_total_messages_stats(sect, stats, previous_stats):
     sect.append(report_nodes.Table(children=lines, cols=4, rheaders=1))
 
 
-def report_messages_stats(sect, stats, _):
+def report_messages_stats(
+    sect,
+    stats: CheckerStatistics,
+    _: CheckerStatistics,
+):
     """make messages type report"""
     if not stats["by_msg"]:
         # don't print this report when we didn't detected any errors
         raise exceptions.EmptyReportError()
-    in_order = sorted(
+    by_msg_stats: Dict[str, int] = stats["by_msg"]  # type: ignore
+    in_order: List[Tuple[int, str]] = sorted(
         (value, msg_id)
-        for msg_id, value in stats["by_msg"].items()
+        for msg_id, value in by_msg_stats.items()
         if not msg_id.startswith("I")
     )
     in_order.reverse()
-    lines = ("message id", "occurrences")
+    lines: Tuple[str, ...] = ("message id", "occurrences")
```

Comment:
Again, implicit.

Reply:
Similar problem, this time on line 43.
https://github.com/PyCQA/pylint/blob/04706af5d411ef5359fa195565f21d3fef7aed2b/pylint/lint/report_functions.py#L41-L43